test/jdk/javax/swing/JButton/HtmlButtonImageTest/HtmlButtonImageTest.java

```diff
@@ -0,0 +1,182 @@
+                frame.dispose();
+            });
+
+            testImageCentering(leftClr, rightClr, topClr, botClr);
+        }
+
+    }
+
+    private static void createAndShowGUI()
+    {
+        frame = new JFrame();
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        frame.setLayout(new FlowLayout());
+
+        // create JButton of size 37x37 text set to a 19x19 image of a red square loaded through html tags
+        button = new JButton();
+        button.setFocusPainted(false);
+        button.setPreferredSize(new Dimension(BUTTON_WIDTH, BUTTON_HEIGHT));
+        button.setText("<html><img src='" + srcDir.resolve("red_square.png").toUri() + "'></html>");
+
+        frame.add(button);
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    private static void setupCenterCoord() {
+        // adjust coordinates to be the center of the button
+        point = button.getLocationOnScreen();
+        point.x += BUTTON_WIDTH / 2;
+        point.y += BUTTON_HEIGHT / 2;
+    }
+
+    private static boolean checkRedness(Color c) {
+        // checks for redness since anti-aliasing causes edges to be not exactly 255,0,0 rgb values
+        if(c.getRed() > 250 && c.getBlue() < 10 && c.getGreen() < 10) {
+            return true;
+        }
+        return false;
+    }
+
+    private static void testImageCentering(Color left, Color right, Color top, Color bottom) {
+        // check if all colors at each edge of square are red
+        if(!checkRedness(left) || !checkRedness(right) || !checkRedness(top) || !checkRedness(bottom)) {
+            System.out.println("-- Failed");
+//            throw new RuntimeException("HTML image not centered in button" + left + right + top + bottom);
+        }
+        else {
+            System.out.println("-- Passed");
+        }
+    }
+
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+            System.out.println("-- " + laf.getName());
+        } catch (UnsupportedLookAndFeelException ignored){
+            System.out.println("Unsupported LookAndFeel: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException |
+                IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void generateImage() {
+        BufferedImage bImg = new BufferedImage(SQUARE_WIDTH, SQUARE_HEIGHT, BufferedImage.TYPE_INT_RGB);
+        Graphics2D cg = bImg.createGraphics();
+        // paint a red square onto cg
+        cg.setColor(Color.RED);
+        cg.fillRect(0, 0, SQUARE_WIDTH, SQUARE_HEIGHT);
+        try {
+            if (ImageIO.write(bImg, "png", new File(srcDir + "/red_square.png")))
+            {
+                System.out.println("-- Saved Image");
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
If this fails, your test fails, doesn't it ? Might as well propagate the exception and let the test exit  with the IOException ..